ols/utils/token_handler.py

```diff
@@ -81,6 +80,10 @@ def get_available_tokens(self, prompt: str, model_config: ModelConfig) -> int:
         Returns:
             available_tokens: int, tokens that can be used for augmentation.
         """
+        # TODO: Now we have option to set context window & response limit set
+        # from the config. With this we need to change default max token parameter
+        # for the model dynamically. Also a check for
```

Comment:
not clear why it changes the max token parameter or why it needs to change dynamically?  

if we just always set a max generated tokens limit of 500, what is the issue?  What would it mean to set it dynamically?  Would you attempt to set it to however many tokens are remaining in the context window after the prompt/context/history is added?